lib/util/utf8.ml

```diff
@@ -26,6 +26,15 @@ let get s i =
     else loop (next s i) (pred k)
   in loop 0 i
 
+(** [Utf8.nb_utf8_char s n] returns the number of characters (not bytes) before index n
```

Comment:
I would drop `utf8_` since we already are in `Utf8` module (and no other function has `utf8_` in its name) and factorize code like this:

```
(** [nb_char s n] returns the number of characters (not bytes) before index [n]
    in string [s].
*)
let nb_char s n =
  let rec loop i k =
    if i = n then k
    else loop (next s i) (succ k)
  in loop 0 0

(** Return the length (number of characters, not bytes)
    of the given string.
*)
let length s = nb_char s (String.length s)
```

Reply:
done.
I still have a small bug to clear before this PR can be merged!!